liquibase-core/src/main/java/liquibase/command/core/GenerateChangelogCommandStep.java

```diff
@@ -1,78 +1,166 @@
 package liquibase.command.core;
 
+import liquibase.Scope;
 import liquibase.command.*;
+import liquibase.command.core.helpers.DbUrlConnectionCommandStep;
+import liquibase.command.core.helpers.DiffOutputControlCommandStep;
+import liquibase.command.core.helpers.ReferenceDbUrlConnectionCommandStep;
+import liquibase.command.providers.ReferenceDatabase;
 import liquibase.configuration.ConfigurationValueObfuscator;
-import liquibase.exception.CommandExecutionException;
+import liquibase.database.Database;
+import liquibase.database.ObjectQuotingStrategy;
+import liquibase.diff.DiffResult;
+import liquibase.diff.output.DiffOutputControl;
+import liquibase.diff.output.changelog.DiffToChangeLog;
+import liquibase.exception.CommandValidationException;
+import liquibase.resource.PathHandlerFactory;
+import liquibase.resource.Resource;
+import liquibase.util.StringUtil;
 
+import java.io.IOException;
+import java.io.PrintStream;
 import java.util.Arrays;
+import java.util.List;
 
-public class GenerateChangelogCommandStep extends AbstractCliWrapperCommandStep {
+public class GenerateChangelogCommandStep extends AbstractCommandStep {
 
     public static final String[] COMMAND_NAME = {"generateChangelog"};
 
-    public static final CommandArgumentDefinition<String> USERNAME_ARG;
-    public static final CommandArgumentDefinition<String> PASSWORD_ARG;
-    public static final CommandArgumentDefinition<String> URL_ARG;
+    private static final String INFO_MESSAGE =
+            "BEST PRACTICE: When generating formatted SQL changelogs, always check if the 'splitStatements' attribute" + System.lineSeparator() +
+            "works for your environment. See https://docs.liquibase.com/commands/generatechangelog.html for more information. ";
+
+    public static final CommandArgumentDefinition<String> AUTHOR_ARG;
+    public static final CommandArgumentDefinition<String> CONTEXT_ARG;
```

Comment:
What is `CONTEXT_ARG`? I'm guessing this is not the same thing we call `contexts` on the command line but am not clear what it is otherwise.

Reply:
No, it was something internal that already existed before .